drivers/flash/flash_mcux_flexspi_nor.c

```diff
@@ -886,6 +886,37 @@ static int flash_flexspi_nor_check_jedec(struct flash_flexspi_nor_data *data,
 
 	/* Switch on manufacturer and vendor ID */
 	switch (vendor_id & 0xFFFF) {
+	case 0x40ef:
```

Comment:
not going to request changes to this PR about this right now since it's a bug fix, but this doesn't seem scalable here, we dont really want all this code for many different flash chips in the image, do we  ? and do we expect the flexspi hardware connection to change dynamically (runtime) ? ie, why isnt these luts somehow determined statically

Reply:
No, it isn't scalable at all. The SFDP driver design was initially intended to save us from statically defined LUTs, but most flash chips that support 4 byte addressing do it via a "vendor defined command set"- so for example, this W25Q512JV worked fine with 3 byte addressing, but that doesn't let you access all 64 MB of the flash chip, hence the need for this switch case.

Long term the solution is to move the FlexSPI under the MSPI API, so we can just write MSPI drivers for all these flash chips.

Reply:
SFDP has codes for standard instructions but JESD216 but mentions that vendors often chosen to extend set of instructions; there is table of instructions that support 4 byte addressin, but in this case this is standard 0xec Fast Read 1s-4s-4s.

Reply:
I have jesd216.F from 2022, and there is chapter: 6.7 "JEDEC 4-byte Address Instruction Parameter Header and Table", p67. 

You can also search for "Support for (1S-1S-4S) Page Program Command, Instruction=34h"

Reply:
But I think that since you have already limited the setup to specific vendor/memory, there is no need to query and you can just enforce directly what you want.